linera-chain/src/data_types.rs

```diff
@@ -600,15 +600,29 @@ impl<'a> LiteCertificate<'a> {
 }
 
 /// A certified statement from the committee.
-#[derive(Clone, Debug, Serialize)]
-#[cfg_attr(with_testing, derive(Eq, PartialEq))]
-pub struct Certificate {
-    /// The certified value.
-    pub value: HashedCertificateValue,
-    /// The round in which the value was certified.
-    pub round: Round,
-    /// Signatures on the value.
-    signatures: Vec<(ValidatorName, Signature)>,
+pub type Certificate = GenericCertificate<CertificateValue>;
+
+impl Serialize for Certificate {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        #[derive(Debug, Serialize)]
+        #[serde(rename = "Certificate")]
+        struct CertificateHelper {
+            value: CertificateValue,
+            round: Round,
+            signatures: Vec<(ValidatorName, Signature)>,
+        }
+
+        let helper = CertificateHelper {
```

Comment:
The `CertificateHelper` could use references for the value and the signatures, then we don't have to clone them here.

Reply:
Done in [Review part 1](https://github.com/linera-io/linera-protocol/pull/2877/commits/e1348d1fc4206a1912eea5f7deb4f771c5d9084f).